### ISSUE TITLE
pip install failure since setuptools>=58.5.0

### ISSUE THREAD

**Issue by iROCKBUNNY** [Legrandin/pycryptodome#568]
os: macOS 12.0.1
Python 3.8.9

```
  ERROR: Command errored out with exit status 1:
   command: <my_project_dir>/venv/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/p7/z62cw0ld5s768_0zr8b99rrm0000gn/T/pip-install-emqto1xd/pycryptodome_1780e409d7b74fc9ad1c5166b4cc010c/setup.py'"'"'; __file__='"'"'/private/var/folders/p7/z62cw0ld5s768_0zr8b99rrm0000gn/T/pip-install-emqto1xd/pycryptodome_1780e409d7b74fc9ad1c5166b4cc010c/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/p7/z62cw0ld5s768_0zr8b99rrm0000gn/T/pip-pip-egg-info-70djabxi
       cwd: /private/var/folders/p7/z62cw0ld5s768_0zr8b99rrm0000gn/T/pip-install-emqto1xd/pycryptodome_1780e409d7b74fc9ad1c5166b4cc010c/
  Complete output (57 lines):
  Testing support for clang
  Target does support clang
  Testing support for gcc
  Target does not support gcc
  Testing support for stdint.h header
  Target does support stdint.h header
  Testing support for 128-bit integer
  Target does support 128-bit integer
  Testing support for cpuid.h header
  Target does not support cpuid.h header
  Testing support for intrin.h header
  Target does not support intrin.h header
  Testing support for posix_memalign
  Target does support posix_memalign
  Testing support for SSE2(intrin.h)
  Target does not support SSE2(intrin.h)
  Testing support for SSE2(x86intrin.h)
  Target does not support SSE2(x86intrin.h)
  Testing support for SSE2(emmintrin.h)
  Target does not support SSE2(emmintrin.h)
  Warning: compiler does not support AESNI instructions
  Warning: compiler does not support CLMUL instructions
  running egg_info
  creating /private/var/folders/p7/z62cw0ld5s768_0zr8b99rrm0000gn/T/pip-pip-egg-info-70djabxi/pycryptodome.egg-info
  writing /private/var/folders/p7/z62cw0ld5s768_0zr8b99rrm0000gn/T/pip-pip-egg-info-70djabxi/pycryptodome.egg-info/PKG-INFO
  writing dependency_links to /private/var/folders/p7/z62cw0ld5s768_0zr8b99rrm0000gn/T/pip-pip-egg-info-70djabxi/pycryptodome.egg-info/dependency_links.txt
  writing top-level names to /private/var/folders/p7/z62cw0ld5s768_0zr8b99rrm0000gn/T/pip-pip-egg-info-70djabxi/pycryptodome.egg-info/top_level.txt
  writing manifest file '/private/var/folders/p7/z62cw0ld5s768_0zr8b99rrm0000gn/T/pip-pip-egg-info-70djabxi/pycryptodome.egg-info/SOURCES.txt'
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/private/var/folders/p7/z62cw0ld5s768_0zr8b99rrm0000gn/T/pip-install-emqto1xd/pycryptodome_1780e409d7b74fc9ad1c5166b4cc010c/setup.py", line 483, in <module>
      setup(
    File "<my_project_dir>/venv/lib/python3.8/site-packages/setuptools/__init__.py", line 159, in setup
      return distutils.core.setup(**attrs)
    File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/distutils/dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "<my_project_dir>/venv/lib/python3.8/site-packages/setuptools/command/egg_info.py", line 299, in run
      self.find_sources()
    File "<my_project_dir>/venv/lib/python3.8/site-packages/setuptools/command/egg_info.py", line 306, in find_sources
      mm.run()
    File "<my_project_dir>/venv/lib/python3.8/site-packages/setuptools/command/egg_info.py", line 541, in run
      self.add_defaults()
    File "<my_project_dir>/venv/lib/python3.8/site-packages/setuptools/command/egg_info.py", line 578, in add_defaults
      sdist.add_defaults(self)
    File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/distutils/command/sdist.py", line 226, in add_defaults
      self._add_defaults_python()
    File "<my_project_dir>/venv/lib/python3.8/site-packages/setuptools/command/sdist.py", line 113, in _add_defaults_python
      self._add_data_files(self._safe_data_files(build_py))
    File "<my_project_dir>/venv/lib/python3.8/site-packages/setuptools/command/egg_info.py", line 621, in _safe_data_files
      return build_py.get_data_files_without_manifest()
    File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/distutils/cmd.py", line 103, in __getattr__
      raise AttributeError(attr)
  AttributeError: get_data_files_without_manifest
  ----------------------------------------
```

**Comment by jiasli** [comment]
Related to https://github.com/pypa/setuptools/issues/2849

**Comment by Legrandin** [comment]
Fixed in setuptools v58.5.3
